### PR TITLE
enable gcov and merge pipelines

### PR DESCRIPTION
**Pull request by salsal97**
precautionary disabling some additional tests
update machine to 20.04 v3
general linting and formatting

### CODE REVIEW

**Comment by salsal97** [comment]
Successful run here: https://openenclaveci.westus.cloudapp.azure.com/blue/organizations/jenkins/Mystikos%2FCode-Coverage/detail/Code-Coverage/57/pipeline

**Review by radhikaj — APPROVED** [review]

**Comment by radhikaj** [comment]
@salsal97  - Can you open an issue so I dont lose track of the tests that are not enabled currently?

**Comment by salsal97** [comment]
New successful run: https://openenclaveci.westus.cloudapp.azure.com/blue/organizations/jenkins/Mystikos%2FCode-Coverage/detail/Code-Coverage/66/pipeline/78

File: .jenkins/pipelines/code-coverage.Jenkinsfile [hunk 1]
```diff
@@ -5,79 +5,325 @@
...
-        timeout(time: 300, unit: 'MINUTES')
+        timeout(time: 600, unit: 'MINUTES')
         timestamps ()
     }
     parameters {
         string(name: "REPOSITORY", defaultValue: "deislabs/mystikos")
         string(name: "BRANCH", defaultValue: "main", description: "Branch to build")
         choice(name: "REGION", choices:['useast', 'canadacentral'], description: "Azure region for the SQL solutions test")
         string(name: "PULL_REQUEST_ID", defaultValue: "", description: "If you are building a pull request, enter the pull request ID number here. (ex. 789)")
+        string(name: "COMMIT_SYNC", defaultValue: "", description: "optional - used to sync outputs of parallel jobs")
```
Comment (CyanDevs) [inline .jenkins/pipelines/code-coverage.Jenkinsfile:19]: Nit: this parameter is only needed when running sub jobs

File: .jenkins/pipelines/code-coverage.Jenkinsfile [hunk 2]
```diff
@@ -5,79 +5,325 @@
...
+                    branches: [[name: "pr/${PULL_REQUEST_ID}"]],
+                    extensions: [],
+                    userRemoteConfigs: [[
+                        url: 'https://github.com/deislabs/mystikos',
+                        refspec: "+refs/pull/${PULL_REQUEST_ID}/merge:refs/remotes/origin/pr/${PULL_REQUEST_ID}"
+                    ]]
+                ])
+            }
+        }
+        stage('Verify commit sync') {
```
Comment (CyanDevs) [inline .jenkins/pipelines/code-coverage.Jenkinsfile:62]: Nit: you can remove this entire stage for Verify commit sync as this pipeline is running tests serially

File: .jenkins/pipelines/code-coverage.Jenkinsfile [hunk 3]
```diff
@@ -5,79 +5,325 @@
...
+            }
+        }
+        stage('Build repo source') {
+            steps {
+                sh """
+                   ${JENKINS_SCRIPTS}/global/make-world.sh
+                   """
+            }
+        }
+        stage("Install package") {
```
Comment (CyanDevs) [inline .jenkins/pipelines/code-coverage.Jenkinsfile:114]: There are two conflicting stages here -- building Mystikos from source and then installing it from deb/tar. Which one should be done?

File: .jenkins/pipelines/code-coverage.Jenkinsfile [hunk 4]
```diff
@@ -5,79 +5,325 @@
...
+            }
+        }
+        stage('Cleanup') {
+            steps {
+                cleanWs()
+            }
+        }
+    }
+    post {
+        failure {
```
Comment (CyanDevs) [inline .jenkins/pipelines/code-coverage.Jenkinsfile:302]: Nit: when solutions test failures are caught, this pipeline would send two emails for the same job

File: .jenkins/pipelines/code-coverage.Jenkinsfile [hunk 5]
```diff
@@ -5,79 +5,325 @@
...
+                   ${JENKINS_SCRIPTS}/global/wait-dpkg.sh
+                   ${JENKINS_SCRIPTS}/global/init-config.sh
+
+                   # Install global dependencies
+                   ${JENKINS_SCRIPTS}/global/wait-dpkg.sh
+                   ${JENKINS_SCRIPTS}/global/init-install.sh
+                   """
+            }
+        }
+        stage('Setup AZ CLI') {
```
Comment (CyanDevs) [inline .jenkins/pipelines/code-coverage.Jenkinsfile:94]: Is az cli used in this pipeline? If not, this stage can be removed.

File: .jenkins/pipelines/code-coverage.Jenkinsfile (hunk 1 above)
Comment (salsal97) [reply .jenkins/pipelines/code-coverage.Jenkinsfile]: Removed this param

File: .jenkins/pipelines/code-coverage.Jenkinsfile (hunk 2 above)
Comment (salsal97) [reply .jenkins/pipelines/code-coverage.Jenkinsfile]: Removed

File: .jenkins/pipelines/code-coverage.Jenkinsfile (hunk 3 above)
Comment (salsal97) [reply .jenkins/pipelines/code-coverage.Jenkinsfile]: As we discussed, I removed the package stage

File: .jenkins/pipelines/code-coverage.Jenkinsfile (hunk 4 above)
Comment (salsal97) [reply .jenkins/pipelines/code-coverage.Jenkinsfile:258]: I removed the report failures step

File: .jenkins/pipelines/code-coverage.Jenkinsfile (hunk 5 above)
Comment (salsal97) [reply .jenkins/pipelines/code-coverage.Jenkinsfile:74]: It is used in the azure-sdk tests - the pipeline had failed when this stage was removed

**Review by CyanDevs — APPROVED** [review]
Thanks for fixing the code coverage pipeline and reports. There are some optimizations that should be done, but I would just be nitpicking so lets do those next iteration.

File: scripts/myst_cc [hunk 6]
```diff
@@ -38,4 +47,5 @@ rm -rf "${MYST_ROOT}"/lcov
 mkdir "${MYST_ROOT}"/lcov
 genhtml --branch-coverage -o "${MYST_ROOT}"/lcov "${MYST_ROOT}"/lcov.info
 
-python3 -m "lcov_cobertura" lcov.info -o lcov.xml
+wget https://raw.githubusercontent.com/eriwen/lcov-to-cobertura-xml/master/lcov_cobertura/lcov_cobertura.py
```
[the remark this answers is not in the document]
Comment (salsal97) [reply scripts/myst_cc]: Added that

**Pull request merged**